distribution/xfer/download.go

```diff
@@ -33,10 +43,16 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 
 // NewLayerDownloadManager returns a new LayerDownloadManager.
 func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int) *LayerDownloadManager {
-	return &LayerDownloadManager{
-		layerStore: layerStore,
-		tm:         NewTransferManager(concurrencyLimit),
+	ldm := &LayerDownloadManager{
+		layerStore:               layerStore,
+		tm:                       NewTransferManager(concurrencyLimit, filepath.Join(os.Getenv("TMPDIR"), "resumable_layer_downloads")),
+		cachedLayerRefs:          make(map[string]layer.Layer),
+		imagePullCheckpointsPath: filepath.Join(os.Getenv("TMPDIR"), "resumable_image_downloads"),
```

Comment:
Also keep in mind that `/tmp` is cleared after restart on many platforms, so you may prefer to store the cache somewhere else.